src/taskgraph/run-task/run-task

```diff
@@ -1316,12 +1316,9 @@ def main(args):
     resource_process = None
 
     try:
-        for k in ["MOZ_FETCHES_DIR", "UPLOAD_DIR"] + [
-            "{}_PATH".format(repository["project"].upper())
-            for repository in repositories
-        ]:
-            if k in os.environ:
-                os.environ[k] = os.path.abspath(os.environ[k])
+        for k, v in os.environ.items():
+            if "$TASK_WORKDIR" in v:
+                os.environ[k] = v.replace("$TASK_WORKDIR", task_workdir)
```

Comment:
This feels a bit wrong, what about `${TASK_WORKDIR}`? Might be worth generalizing this and using `os.path.expandvar`?

Reply:
Yeah, it's a valid concern. As a user I would definitely expect both to work though.

Reply:
Hmm..
```
>>> os.path.expandvars("$HOME/$FOO/$BAR")
'/home/ahal/$FOO/$BAR'
```

so it does leave things unchanged if the env doesn't exist... Maybe that's ok then?

Reply:
Alright, I don't want to get too distracted here, and this change would potentially be a big distraction. I'll update the PR to support both `$TASK_WORKDIR` and `${TASK_WORKDIR}`, then file an issue to explore making it more generic.